src/compiler/factory/utilities.ts

```diff
@@ -700,9 +700,10 @@ export function createExternalHelpersImportDeclarationIfNeeded(nodeFactory: Node
         const impliedModuleKind = getImpliedNodeFormatForEmitWorker(sourceFile, compilerOptions);
         const helpers = getImportedHelpers(sourceFile);
         if (
-            (moduleKind >= ModuleKind.ES2015 && moduleKind <= ModuleKind.ESNext) ||
-            impliedModuleKind === ModuleKind.ESNext ||
-            impliedModuleKind === undefined && moduleKind === ModuleKind.Preserve
+            impliedModuleKind !== ModuleKind.CommonJS &&
+            ((moduleKind >= ModuleKind.ES2015 && moduleKind <= ModuleKind.ESNext) ||
+                impliedModuleKind === ModuleKind.ESNext ||
+                impliedModuleKind === undefined && moduleKind === ModuleKind.Preserve)

```

Comment:
I was able to simplify this condition in Corsa since we’re removing System, AMD, and UMD. Opting to make the minimal change here rather than try to clean up.